avocado/plugins/runner.py

```diff
@@ -348,26 +348,23 @@ def _iter_suite(self, job, test_suite, variants, execution_order):
             raise NotImplementedError("Suite_order %s is not supported"
                                       % execution_order)
 
-    def run_suite(self, job, result, test_suite, variants, timeout=0,
-                  replay_map=None, execution_order=None):
+    def run_suite(self, job, result, test_suite, variants, config):
         """
         Run one or more tests and report with test result.
 
         :param job: an instance of :class:`avocado.core.job.Job`.
         :param result: an instance of :class:`avocado.core.result.Result`
         :param test_suite: a list of tests to run.
         :param variants: A varianter iterator to produce test params.
-        :param timeout: maximum amount of time (in seconds) to execute.
-        :param replay_map: optional list to override test class based on test
-                           index.
-        :param execution_order: Mode in which we should iterate through tests
-                                and variants.  If not provided, will default to
-                                :attr:`DEFAULT_EXECUTION_ORDER`.
+        :param config: config dict to run the suite.
         :return: a set with types of test failures.
         """
         summary = set()
+        timeout = config.get('run.job_timeout')
+        replay_map = config.get('replay_map')
+        execution_order = config.get('run.execution_order')
         queue = multiprocessing.SimpleQueue()
-        if timeout > 0:
+        if timeout is not None and timeout > 0:
```

Comment:
If `timeout` default is registered with 0, do we need to check if it is `None` here?

Reply:
Yes, I had that impression too. But for some reason it is None. I will do a double-check.

Reply:
Thanks again for that @willianrampazzo . Actually, we don't need the `config` argument, because `job` has already the `config` (`job.config`) and also this show-me that we need a new refactoring to make timeout registered at `register_job_options()`. :)

Reply:
@willianrampazzo I force pushed here, please double-check if it is ok now.